src/DurableTask.Core/TaskOrchestrationContext.cs

```diff
@@ -638,6 +638,11 @@ public void FailOrchestration(Exception failure)
                     details = orchestrationFailureException.Details;
                 }
             }
+            else if (failure is TaskFailedException tfe && this.ErrorPropagationMode == ErrorPropagationMode.UseFailureDetails)
+            {
+                // Propagate the original FailureDetails
+                failureDetails = tfe.FailureDetails;
+            }
```

Comment:
nit - I think we should avoid single letter 3 word abbreviations :-)
```suggestion
            else if (failure is TaskFailedException taskFailedException && this.ErrorPropagationMode == ErrorPropagationMode.UseFailureDetails)
            {
                // Propagate the original FailureDetails
                failureDetails = taskFailedException .FailureDetails;
            }
```

Reply:
This is common Go conventions (simple local variable names) sneaking into my C# code.